Modules/_pickle.c

```diff
@@ -1526,11 +1557,15 @@ _Unpickler_MemoGet(UnpicklerObject *self, size_t idx)
 /* Returns -1 (with an exception set) on failure, 0 on success.
    This takes its own reference to `value`. */
 static int
-_Unpickler_MemoPut(UnpicklerObject *self, size_t idx, PyObject *value)
+_Unpickler_MemoPut(PickleState *st, UnpicklerObject *self, size_t idx, PyObject *value)
 {
     PyObject *old_item;
 
     if (idx >= self->memo_size) {
+        if (idx > self->memo_len * 2 + (1 << 17)) {
```

Comment:
where does (1 << 17) come from?  that feels like it should be a constant or at least deserves a command as to what the significance of this value is and why this limit exists.